src/result.rs

```diff
@@ -17,6 +17,7 @@ pub type z_result_t = i8;
 
 pub const Z_CHANNEL_DISCONNECTED: z_result_t = 1;
 pub const Z_CHANNEL_NODATA: z_result_t = 2;
+pub const Z_CHANNEL_GENERIC: z_result_t = 3;
```

Comment:
Looks like this should be an error (i.e. Z_E... with negative code), since it is returned in the cases where it something is wrong.